cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/RecordWriter.java

```diff
@@ -380,28 +382,69 @@ private Map<String, Object> getBindValuesForColumns(Map<String, Object> map, Str
     {
         Preconditions.checkArgument(values.length == columnNames.length,
                                     "Number of values does not match the number of columns " + values.length + ", " + columnNames.length);
+
         for (int i = 0; i < columnNames.length; i++)
         {
-            map.put(columnNames[i], maybeConvertUdt(values[i]));
+            if (cqlTable().containsUdt(columnNames[i]))
+            {
+                map.put(columnNames[i], maybeConvertUdt(values[i]));
+            }
+            else
+            {
+                map.put(columnNames[i], values[i]);
+            }
         }
         return map;
     }
 
     private Object maybeConvertUdt(Object value)
```

Comment:
It would be nice to add java doc to describe the method, e.g. why it checks for collections in the implementation.